adx/resource_adx_table_retention_policy.go

```diff
@@ -0,0 +1,121 @@
+package adx
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"encoding/json"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type TableRetentionPolicy struct {
+	SoftDeletePeriod string
+	Recoverability string
+}
+
+func resourceADXTableRetentionPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceADXTableRetentionPolicyCreate,
+		ReadContext:   resourceADXTableRetentionPolicyRead,
+		DeleteContext: resourceADXTableRetentionPolicyDelete,
+		UpdateContext: resourceADXTableRetentionPolicyCreate,
```

Comment:
At least by azurerm convention - if we use the same func to do both Create and Update - func name is `resourceXXXCreateUpdate`. Could we amend that please?

Reply:
fixed